package/plato/package

```diff
@@ -62,12 +62,56 @@ package() {
 configure() {
     local mediadir=/home/root/plato-media
 
-    # This directory should not be tracked by opkg
-    # so that its contents are kept after removing
     if [ ! -d $mediadir ]; then
+        # This directory should not be tracked by opkg
+        # so that its contents are kept after removing
         mkdir $mediadir
         echo ""
         echo "Place your media files for plato in '$mediadir'"
         echo ""
+    else
+        # Fix breaking change introduced in version 0.9.14
+        # https://github.com/baskerville/plato/releases/tag/0.9.14
+
+        local metadata_file=$mediadir/.metadata.json
+        local metadata_file_old=$mediadir/.metadata-old.json
+
+        local max_key_len
+        [ -e $metadata_file ] && max_key_len=$(jq --raw-output 'keys[]' < $metadata_file | wc -L) || max_key_len=-1
+        if [ "$max_key_len" -gt 16 ] && [ ! -e $metadata_file_old ]; then
+
+            echo ""
+            echo "In Plato 0.9.14 the metadata format for libraries changed."
+            echo "Found your outdated metadata file. Will update it automatically..."
+
+            # Routine is based on baskerville's migrate-hex_keys.py (from the release page)
+            # Rewritten in bash to only require jq rather than python.
+            mv $metadata_file $metadata_file_old
+            local change_cmds=""
+            local del_list=""
+            for key in $(jq --raw-output 'keys[]' < $metadata_file_old); do
+                hex_key=$(printf %016X "$key")
+                #echo "Key: $key --> $hex_key"
+
+                [ -z "$del_list" ] || del_list="${del_list}, "
+                [ -z "$change_cmds" ] || change_cmds="${change_cmds} | "
+                del_list="${del_list}.[\"${key}\"]"
+                change_cmds="${change_cmds}.[\"$hex_key\"] = .[\"$key\"]"
+            done
+
+            # Create updated json file
+            jq "$change_cmds | del($del_list)" < $metadata_file_old > $metadata_file
+            echo "Update complete. The old file was renamed to '.metadata-old.json'."
+            echo ""
+
+        elif [ "$max_key_len" -gt 16 ] && [ -e $metadata_file_old ] || [ ! -e $metadata_file ]; then
+            echo ""
+            echo "In Plato 0.9.14 the metadata format for libraries changed."
+            echo "MANUAL INTERVENTION REQUIRED\!\!\!"
+            echo "Please follow the following guide to update your library metadata manually:"
+            echo "https://github.com/baskerville/plato/releases/tag/0.9.14"
+            echo "Please LAUNCH plato ONLY AFTER this was done\!"
```

Comment:
i didn't go through this, i just blew it away and restarted. i think this should be done automatically (since it makes a backup, i think?) and the user notified

Reply:
I agree, this should be done automagically by the package. It's one of the use-cases for pre/post upgrade scripts.

Reply:
I used configure as I feared pre/post upgrade might miss cases where a user fully reinstalled a package. The script checks for old versions of the metadata and I attempted to make it check more often/reliably this way.

I can change this tomorrow though. Is still a good point.

Reply:
i think i'm a bit confused. when i upgraded plato, i got the message (`MANUAL INTERVENTION REQUIRED`), but it looks like you automatically do the migration for people. Did I get that message because I didn't have any metadata?

Reply:
I think this happened because you seem like you didn't do the migration but also have a `.old` (backup) file. I just didn't support that usecase and added that hint instead of trying to deal with it.

Reply:
Also just looked at the [docs](https://github.com/toltec-dev/toltec/blob/stable/docs/package.md#install-section):

> When upgrading a package from version A to B, the following happens:
> 
>  - `preupgrade B`, if it exists, is called from version A
>  -  Old package files are removed (except configuration files)
>  -  `postupgrade B`, if it exists, is called from version A
>  -  New package files are unpacked and installed
>  -  `configure`, if it exists, is called from version B

If I read this correctly, calling postupgrade wouldn't make sense since it would only call if this package ever gets upgraded to a future one.